http/oauth2/oauth2_test.go

```diff
@@ -80,6 +81,143 @@ func TestHandlerIntrospectError(t *testing.T) {
 	}
 }
 
+type tokenIntrospectedSuccessful struct {
+	response *IntrospectResponse
+}
+
+func (t *tokenIntrospectedSuccessful) IntrospectToken(ctx context.Context, token string) (*IntrospectResponse, error) {
+	return t.response, nil
+}
+
+func TestAuthenticatorWithSuccess(t *testing.T) {
+	testCases := []struct {
+		desc string
+		auth *Authorizer
+	}{
+		{desc: "Tests a valid Request with OAuth2 Authentication without Scope checking",
+			auth: NewAuthenticator(&tokenIntrospectedSuccessful{&IntrospectResponse{
+				Active:   true,
+				Scope:    "ABC DHHG kjdk",
+				ClientID: "ClientId",
+				UserID:   "UserId",
+			}}, &Config{}),
+		},
+		{desc: "Tests a valid Request with OAuth2 Authentication with Scope checking",
+			auth: NewAuthenticator(&tokenIntrospectedSuccessful{&IntrospectResponse{
+				Active:   true,
+				Scope:    "ABC DHHG kjdk",
+				ClientID: "ClientId",
+				UserID:   "UserId",
+			}}, &Config{}).WithScope("ABC"),
+		},
+	}
+	for _, tC := range testCases {
+		t.Run(tC.desc, func(t *testing.T) {
+			w := httptest.NewRecorder()
+			r := httptest.NewRequest("GET", "/", nil)
+			r.Header.Add("Authorization", "Bearer bearer")
+			authorize, b := tC.auth.Authorize(r, w)
+
+			if !b || w.Code != http.StatusOK || authorize == nil {
+				t.Errorf("Expected succesfull Authentication, but was not succesfull with code %d and body %v", w.Code, w.Body)
+				return
+			}
+			to, _ := security.GetTokenFromContext(authorize)
+			tok, ok := to.(*token)
+
+			if !ok || tok.value != "bearer" || tok.scope != Scope("ABC DHHG kjdk") || tok.clientID != "ClientId" || tok.userID != "UserId" {
```

Comment:
If you were to test all these values, you should restructure the test cases. For example define the scope, client id and user id as attributes of the test case and then create the authenticator inside the test (like you did in `TestAuthenticationWithErrors`). Because if I would add a test case that did not exactly have these scope, client id and user id, it would not work.